File: frontend/lib/domain/model/workout/workout_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:workoutride/domain/model/workout/workout_block_result.dart';

part 'workout_result.freezed.dart';

@freezed
class WorkoutResult with _$WorkoutResult {
  const factory WorkoutResult({
    required int id,
    required int workoutSummaryId,
    required DateTime startedAt,
    DateTime? finishedAt,
    required int totalDurationSeconds,
    int? averagePower,
    int? maxPower,
    int? averageCadence,
    required String status,
    required DateTime createdAt,
    required DateTime updatedAt,
    @Default([]) List<WorkoutBlockResult> workoutBlockResults,
  }) = _WorkoutResult;
}

File: backend/app/api/entities/workout_result.rb
module Entities
  class WorkoutResult < Grape::Entity
    expose :id
    expose :workout_summary_id
    expose :started_at
    expose :finished_at
    expose :total_duration_seconds
    expose :average_power
    expose :max_power
    expose :average_cadence
    expose :status
    expose :created_at
    expose :updated_at
    expose :workout_block_results, using: Entities::WorkoutBlockResult
  end
end
